lnbits/core/services.py

```diff
@@ -167,18 +164,18 @@ async def create_invoice(
 
     invoice = bolt11_decode(payment_request)
 
-    amount_msat = 1000 * amount_sat
     await create_payment(
         wallet_id=wallet_id,
         checking_id=checking_id,
         payment_request=payment_request,
         payment_hash=invoice.payment_hash,
-        amount=amount_msat,
+        amount=1000 * amount_sat,
         expiry=invoice.expiry_date,
         memo=memo,
         extra=extra,
         webhook=webhook,
         conn=conn,
+        fee=abs(fees_msats or 0),
```

Comment:
using `abs()` feels dirty

what do you think @dni? are there funding sources that return negative fee and some funding sources that return positive fee? if that is the case, we should fix the funding sources to always return positive fee.

Reply:
i think its ok for now, tried to tackle it here, but somehow i broke regtest and couldnt relally figure it out. https://github.com/lnbits/lnbits/pull/2468 going to give it another try sometime